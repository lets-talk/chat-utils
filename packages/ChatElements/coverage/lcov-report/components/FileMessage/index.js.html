<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for components/FileMessage/index.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> / <a href="index.html">components/FileMessage</a> index.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">45.45% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>5/11</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">11.11% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>2/18</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">66.67% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>2/3</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">50% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>5/10</span>
      </div>
    </div>
  </div>
  <div class='status-line low'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74
75
76
77
78
79
80
81
82
83
84
85
86
87
88</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
&nbsp;
import FaCloudDownload from 'react-icons/lib/fa/cloud-download';
import FaFile from 'react-icons/lib/fa/file';
&nbsp;
import './index.scss';
&nbsp;
export class FileMessage extends Component {
  constructor(props) {
    super(props);
    this.onClick = this.onClick.bind(this);
  }
&nbsp;
<span class="fstat-no" title="function not covered" >  on</span>Click() {
<span class="cstat-no" title="statement not covered" >    if (!this.props.data.status) { <span class="cstat-no" title="statement not covered" >return; </span>}</span>
&nbsp;
<span class="cstat-no" title="statement not covered" >    if (!this.props.data.status.download &amp;&amp; this.props.onDownload instanceof Function) {</span>
<span class="cstat-no" title="statement not covered" >      this.props.onDownload();</span>
    } else <span class="cstat-no" title="statement not covered" >if (this.props.data.status.download &amp;&amp; this.props.onOpen instanceof Function) {</span>
<span class="cstat-no" title="statement not covered" >      this.props.onOpen();</span>
    }
  }
&nbsp;
  render() {
    return (
      &lt;div className={classNames('letstalk-mbox-attachment', 'letstalk-mbox-file')}&gt;
        &lt;button onClick={this.onClick}&gt;
          &lt;div className="letstalk-mbox-file--icon"&gt;
            &lt;FaFile
              color="#aaa"
            /&gt;
            &lt;div className="letstalk-mbox-file--size"&gt;
              {this.props.data.size}
            &lt;/div&gt;
          &lt;/div&gt;
          &lt;div className="letstalk-mbox-file--text"&gt;
            {this.props.text}
          &lt;/div&gt;
          &lt;div className="letstalk-mbox-file--buttons"&gt;
            {
              this.props.data.status &amp;&amp;
<span class="branch-1 cbranch-no" title="branch not covered" >                !this.props.data.status.download </span>&amp;&amp;
<span class="branch-2 cbranch-no" title="branch not covered" >                !this.props.data.status.click </span>&amp;&amp;
<span class="branch-3 cbranch-no" title="branch not covered" >                &lt;FaCloudDownload</span>
                  color="#aaa"
                /&gt;
            }
            {
              this.props.data.status &amp;&amp;
<span class="branch-1 cbranch-no" title="branch not covered" >                typeof this.props.data.status.loading === 'number' </span>&amp;&amp;
<span class="branch-2 cbranch-no" title="branch not covered" >                this.props.data.status.loading !== 0 </span>&amp;&amp;
<span class="branch-3 cbranch-no" title="branch not covered" >                &lt;div /&gt;</span>
            }
          &lt;/div&gt;
        &lt;/button&gt;
      &lt;/div&gt;
    );
  }
}
&nbsp;
FileMessage.propTypes = {
  text: PropTypes.string,
  data: PropTypes.object,
  onClick: PropTypes.func,
  onDownload: PropTypes.func,
  onOpen: PropTypes.func,
};
&nbsp;
FileMessage.defaultProps = {
  text: '',
  data: {},
  onClick: null,
  onDownload: null,
  onOpen: null,
};
&nbsp;
&nbsp;
// &lt;CircularProgress
//   mode="determinate"
//   value={this.props.data.status.loading}
//   size={80}
//   thickness={5}
// /&gt;
&nbsp;
export default FileMessage;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Fri Nov 24 2017 01:32:25 GMT-0300 (-03)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
</body>
</html>
