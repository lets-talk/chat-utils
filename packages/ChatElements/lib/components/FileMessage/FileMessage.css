.letstalk-mbox-file {
    padding-bottom: 13px;
}

.letstalk-mbox-file > button {
    background: #e9e9e9;
    display: flex;
    border-radius: 5px;
    margin-top: -3px;
    margin-right: -6px;
    margin-left: -6px;
    align-items: center;
    min-height: 52px;
    max-width: 500px;
    padding: 5px 0;
    cursor: pointer;
    user-select: none;
    outline: none;
    border:none;
}

.letstalk-mbox-file > button > * {
    padding: 0px 10px;
}

.letstalk-mbox-file--icon {
    font-size: 30px;
    align-items: center;
    display: flex;
    flex-direction: column;
}

.letstalk-mbox-file--size {
    font-size: 10px;
    margin-top: 3px;
    max-width: 52px;
    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;
}

.letstalk-mbox-file--text {
    font-size: 13.6px;
    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;
}

.letstalk-mbox-file--buttons {
    font-size: 30px;
    align-items: center;
    display: flex;
}

.letstalk-mbox-file--loading {
    font-size: 15px;
}
