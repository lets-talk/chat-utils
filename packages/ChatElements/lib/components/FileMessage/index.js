import React, { Component } from 'react';
import PropTypes from 'prop-types';

import FaCloudDownload from 'react-icons/lib/fa/cloud-download';
import FaFile from 'react-icons/lib/fa/file';
import CircularProgress from 'material-ui/CircularProgress';

import './index.css';

export class FileMessage extends Component {
  constructor(props) {
    super(props);
    this.onClick = this.onClick.bind(this);
  }

  onClick() {
    if (!this.props.data.status) { return; }

    if (!this.props.data.status.download && this.props.onDownload instanceof Function) {
      this.props.onDownload();
    } else if (this.props.data.status.download && this.props.onOpen instanceof Function) {
      this.props.onOpen();
    }
  }

  render() {
    return (
      <div className="letstalk-mbox-file">
        <button onClick={this.onClick}>
          <div className="letstalk-mbox-file--icon">
            <FaFile
              color="#aaa"
            />
            <div className="letstalk-mbox-file--size">
              {this.props.data.size}
            </div>
          </div>
          <div className="letstalk-mbox-file--text">
            {this.props.text}
          </div>
          <div className="letstalk-mbox-file--buttons">
            {
              this.props.data.status &&
                !this.props.data.status.download &&
                !this.props.data.status.click &&
                <FaCloudDownload
                  color="#aaa"
                />
            }
            {
              this.props.data.status &&
                typeof this.props.data.status.loading === 'number' &&
                this.props.data.status.loading !== 0 &&
                <CircularProgress
                  mode="determinate"
                  value={this.props.data.status.loading}
                  size={80}
                  thickness={5}
                />
            }
          </div>
        </button>
      </div>
    );
  }
}

FileMessage.propTypes = {
  text: PropTypes.string,
  data: PropTypes.object,
  onClick: PropTypes.func,
  onDownload: PropTypes.func,
  onOpen: PropTypes.func,
};

FileMessage.defaultProps = {
  text: '',
  data: {},
  onClick: null,
  onDownload: null,
  onOpen: null,
};


export default FileMessage;
