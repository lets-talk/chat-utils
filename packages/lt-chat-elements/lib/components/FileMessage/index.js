import React, { Component } from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import FaCloudDownload from 'react-icons/lib/fa/cloud-download';
import FaFile from 'react-icons/lib/fa/file';

const StyledMessageBox = styled.div`
  padding-bottom: 13px;

  button {
      background: #e9e9e9;
      display: flex;
      border-radius: 5px;
      margin-top: -3px;
      margin-right: -6px;
      margin-left: -6px;
      align-items: center;
      min-height: 52px;
      max-width: 500px;
      padding: 5px 0;
      cursor: pointer;
      user-select: none;
      outline: none;
      border:none;
  }
`;

const StyledFileIcon = styled.div`
  font-size: 30px;
  align-items: center;
  display: flex;

  div {
    font-size: 10px;
    margin-top: 3px;
    max-width: 52px;
    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;
  }
`;

const StyledFileName = styled.div`
  font-size: 13.6px;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
`;

const StyledFileDownloadButtons = styled.div`
  font-size: 30px;
  align-items: center;
  display: flex;
`;

export class FileMessage extends Component {
  constructor(props) {
    super(props);
    this.onClick = this.onClick.bind(this);
  }

  onClick() {
    if (!this.props.data.status) { return; }
    if (!this.props.data.status.download && this.props.onDownload) {
      this.props.onDownload();
    } else if (this.props.data.status.download && this.props.onOpen) {
      this.props.onOpen();
    }
  }

  render() {
    return (
      <StyledMessageBox>
        <button onClick={this.onClick}>
          <StyledFileIcon>
            <FaFile
              color="#aaa"
            />
            <div>
              {this.props.data.size}
            </div>
          </StyledFileIcon>
          <StyledFileName>
            {this.props.text}
          </StyledFileName>
          <StyledFileDownloadButtons>
            {
              this.props.data.status &&
                !this.props.data.status.download &&
                !this.props.data.status.click &&
                <FaCloudDownload
                  color="#aaa"
                />
            }
            {
              this.props.data.status &&
                typeof this.props.data.status.loading === 'number' &&
                this.props.data.status.loading !== 0 &&
                <div />
            }
          </StyledFileDownloadButtons>
        </button>
      </StyledMessageBox>
    );
  }
}

FileMessage.propTypes = {
  text: PropTypes.string,
  data: PropTypes.object,
  onDownload: PropTypes.func,
  onOpen: PropTypes.func,
};

FileMessage.defaultProps = {
  text: '',
  data: {},
  onDownload: null,
  onOpen: null,
};


// <CircularProgress
//   mode="determinate"
//   value={this.props.data.status.loading}
//   size={80}
//   thickness={5}
// />

export default FileMessage;
