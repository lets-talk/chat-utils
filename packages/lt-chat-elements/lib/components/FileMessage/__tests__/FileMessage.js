import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import FileMessage from '../index';

describe('FileMessage component', () => {
  const props = {
    data: {
      uri: 'https://pbs.twimg.com/profile_images/718588760003383296/2AG8omMO_400x400.jpg',
      status: {
        click: true,
        loading: 70,
        download: false,
      },
      width: 300,
      height: 300,
      size: '2Mb',
    },
  };

  it('should render without issues', () => {
    const component = shallow(<FileMessage />);

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should render with data props', () => {
    const component = shallow(<FileMessage {...props} />);

    expect(component.length).toBe(1);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should call the onDownload action', () => {
    const mockOnDownload = jest.fn();
    const fakeEventObject = { preventDefault() {}, stopPropagation() {} };

    const component = shallow(<FileMessage onDownload={mockOnDownload} {...props} />);

    component.find('StyledFileDownloadButton').simulate('click', fakeEventObject);
    expect(mockOnDownload).toHaveBeenCalledTimes(1);
  });

  it('should call the onOpen action', () => {
    const mockOnOpen = jest.fn();
    const fakeEventObject = { preventDefault() {}, stopPropagation() {} };
    const customProps = {
      ...props,
      data: {
        ...props.data,
        status:
          {
            ...props.data.status,
            download: true,
          },
      },
    };

    const component = shallow(<FileMessage onOpen={mockOnOpen} {...customProps} />);
    component.find('StyledFileDownloadButton').simulate('click', fakeEventObject);
    expect(mockOnOpen).toHaveBeenCalledTimes(1);
  });
});
