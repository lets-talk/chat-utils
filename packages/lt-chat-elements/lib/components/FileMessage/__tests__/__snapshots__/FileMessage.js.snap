// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FileMessage component should render with data props 1`] = `
<styled.div
  className="LT-FileMessage-Container"
>
  <button
    onClick={[Function]}
  >
    <styled.div>
      <FaFile
        color="#aaa"
      />
      <div>
        2Mb
      </div>
    </styled.div>
    <styled.div />
    <styled.div>
      <div />
    </styled.div>
  </button>
</styled.div>
`;

exports[`FileMessage component should render without issues 1`] = `
<styled.div
  className="LT-FileMessage-Container"
>
  <button
    onClick={[Function]}
  >
    <styled.div>
      <FaFile
        color="#aaa"
      />
      <div />
    </styled.div>
    <styled.div />
    <styled.div />
  </button>
</styled.div>
`;
